grease/src/Grease/Macaw/ResolveCall.md

```diff
@@ -162,101 +169,180 @@ lookupFunctionHandle bak la halloc arch memory symMap pltStubs dynFunMap funOvs
-                 -> do doLog la $ Diag.PltCall pltStubName funcAddrOff pltCallAddr
-                       dispatchFuncAddrOff pltCallAddr
-                 |  otherwise
-                 -> case Map.lookup pltStubName funOvs of
-                      -- ...otherwise, if there is an override for the PLT stub,
-                      -- use it...
-                      Just macawFnOv -> do
-                        logFunctionCall pltStubName
-                        useMacawFunctionOverride bak la halloc arch funOvs macawFnOv st
-                      -- ...otherwise, skip the PLT call entirely.
-                      Nothing -> do
-                        logFunctionCall pltStubName
-                        skipExternalCall (PltNoOverride @arch funcAddrOff pltStubName)
-
-            -- Finally, check if this is a function that we should explore, and if
-            -- so, use Macaw's code discovery to do so. See Note [Incremental code
-            -- discovery] in Grease.Macaw.SimulatorState.
-            --
-            -- As a simple heuristic for whether a function is worthy of
-            -- exploration, we check if the segment that the address inhabits is
-            -- executable. This check is important to prevent simulating functions
-            -- that, say, inhabit the .data section (which is common for binaries
-            -- that fail the `in-text` requirement), as Macaw will simply crash
-            -- when simulating them.
-            | Discovery.isExecutableSegOff funcAddrOff = do
-              (hdl, st') <-
-                discoverFuncAddr la halloc arch memory symMap pltStubs funcAddrOff st
-              logFunctionCall (C.handleName hdl)
-              callHandle hdl st'
-
-            | otherwise =
-              skipExternalCall (NotExecutable funcAddrOff)
-
-      in case Loader.resolveAbsoluteAddress memory bvMemWord of
-        Nothing -> skipExternalCall (InvalidAddress (BV.ppHex knownNat bv))
-        Just funcAddrOff -> dispatchFuncAddrOff funcAddrOff
+        else skip SymbolicAddress
+```
+</details>
+
+Next, the absolute address is resolved to an address in the binary. If this
+resolution fails, the call is skipped.
+<!-- TODO: Is that right? -->
+
+<details>
+<summary>Skipping calls to invalid addresses</summary>
+
+```haskell
+    Just bv -> do
+      -- This conversion is safe iff MC.ArchAddrWidth arch <= 64
+      let bvWord64 = fromIntegral @Integer @Word64 (BV.asUnsigned bv)
+      let bvMemWord = EL.memWord bvWord64
+      case Loader.resolveAbsoluteAddress memory bvMemWord of
+        -- TODO: This should probably be a hard error (at least by default)?
+        Nothing -> skip (InvalidAddress (ppMemWord bvMemWord))
+        Just funcAddrOff ->
+          callConcreteAddr bak la halloc arch memory symMap pltStubs dynFunMap funOvs st mem regs bvMemWord funcAddrOff
+  where
+    ppMemWord = BV.ppHex knownNat . BV.word64 . EL.memWordValue
+```
+</details>
+  
+<details>
+<summary>TODO</summary>
```

Comment:
If we want to proceed with this approach, I'd document the body of this function similarly to how `lookupFunctionHandle` is described above.